server/app.js

```diff
@@ -81,8 +82,14 @@ app.post('/signup', async (req, res, next) => {
       return next(new BadRequestError('Username already exists. Please try again.'))
     }
 
-    const newUser = await new User({ username, password, location, privacy })
-    await newUser.save()
+    /* randomly decide which similarity method user receives */
```

Comment:
Can you explain this portion to me? 

How we assign a user to a similarityMethod should not be random. Otherwise if you build an experiment you cannot analyze the results since you don't have determinism.

Reply:
Yes, I updated this in the next PR.